src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp

```diff
@@ -734,27 +752,42 @@ void Transformations::MainSnippets(void) {
     tokenization_config.concurrency = config.streamExecutorConfig.get_threads_per_stream();
     if (tokenization_config.concurrency == 0)
         tokenization_config.concurrency = parallel_get_max_threads();
+#if defined(OPENVINO_ARCH_ARM64)
+    // ARM has 32 gprs. After excluding 2 registers for work amounts, 1 register for runtime parameters, 1 platform register,
+    // 3 registers for temporary use, and 2 stack related registers, it has 23 remaining registers.
+    tokenization_config.data_ptr_grp_count = 23;
+#else
+    // X64 has 16 gprs. After excluding 2 registers for work amounts, 1 register for runtime parameters,
+    // and 2 stack related registers, it has 11 remaining registers.
+    tokenization_config.data_ptr_grp_count = 11;
+#endif
     // The optimization "SplitDimensionM" depends on target machine (thread count).
     // To avoid uncontrolled behavior in tests, we disabled the optimization when there is Config::SnippetsMode::IgnoreCallback
-    tokenization_config.split_m_dimension = snippetsMode != Config::SnippetsMode::IgnoreCallback;
+    tokenization_config.split_m_dimension = !ignoreCallback;
     // [122706] Some 3D MHA Patterns have perf regressions when Transpose op is tokenized
     tokenization_config.mha_supported_transpose_ranks = { 4 };
 
     ov::pass::Manager snippetsManager;
     snippetsManager.set_per_pass_validation(false);
-    if (snippetsMode != Config::SnippetsMode::IgnoreCallback)
-        CPU_REGISTER_PASS_X64(snippetsManager, SnippetsMarkSkipped, inferencePrecision != ov::element::f32);
+    if (!ignoreCallback)
+        CPU_REGISTER_PASS_COMMON(snippetsManager, SnippetsMarkSkipped, inferencePrecision != ov::element::f32);
```

Comment:
The transformation `SnippetsMarkSkipped` is used to [mark](https://github.com/openvinotoolkit/openvino/blob/326200598ae346eab72460b65f2fa748b22e8f74/src/plugins/intel_cpu/src/transformations/snippets/x64/pass/snippets_mark_skipped.cpp#L574) some Elementwise ops to not be tokenized by Snippets. For example, it's needed for potential post-ops Elementwise ops.
All of these checks in the pass are described for x64.
However, not all of them can be used for aarch64 too.
For example, [FuseConvolutionSumAndConvolutionSumActivation](https://github.com/openvinotoolkit/openvino/blob/326200598ae346eab72460b65f2fa748b22e8f74/src/plugins/intel_cpu/src/graph_optimizer.cpp#L1685) is supported only x64 - it means that the corresponding [check](https://github.com/openvinotoolkit/openvino/blob/326200598ae346eab72460b65f2fa748b22e8f74/src/plugins/intel_cpu/src/transformations/snippets/x64/pass/snippets_mark_skipped.cpp#L356) in `SnippetsMarkSkipped` should return false for non-x64.
Also there are some conditions in `node::canFuse` ([example](https://github.com/openvinotoolkit/openvino/blob/326200598ae346eab72460b65f2fa748b22e8f74/src/plugins/intel_cpu/src/nodes/eltwise.cpp#L3044) for eltwise). Need to say that there are many conditions and limitations that should be covered by `SnippetsMarkSkipped` for different archs.

Reply:
Will apply arm specific SnippetsMarkSkipped. Thanks Alexandra and Ivan!

Reply:
ARM specific SnippetsMarkSkipped is added according to fixing performance issues. Thanks Alexandra and Ivan!